projects/quml-library/src/lib/player/player.component.ts

```diff
@@ -231,29 +231,54 @@ export class PlayerComponent implements OnInit, AfterViewInit {
     let updated = false;
     if (this.optionSelectedObj !== undefined) {
       let key: any = await this.utilService.getKeyValue(Object.keys(this.questions[currentIndex].responseDeclaration));
-      const correctOptionValue = this.questions[currentIndex].responseDeclaration[key].correctResponse.value;
       this.currentQuestion = this.questions[currentIndex].body;
-      this.currentOptions = this.questions[currentIndex].interactions.response1.options;
-      if (Boolean(option.option.value == correctOptionValue)) {
-        this.currentScore = this.getScore(currentIndex , key);
-        this.showAlert = true;
-        this.alertType = true;
-        this.updateScoreBoard(currentIndex + 1, 'attempted', selectedOptionValue , this.currentScore);
-        this.correctFeedBackTimeOut();
-        if (!this.showFeedBack) {
-          this.nextSlide();
+      this.currentOptions = this.questions[currentIndex].interactions[key].options;
+      if (option.cardinality === 'single') {
+        const correctOptionValue = this.questions[currentIndex].responseDeclaration[key].correctResponse.value;
+        if (Boolean(option.option.value == correctOptionValue)) {
+          this.currentScore = this.getScore(currentIndex, key);
+          this.showAlert = true;
+          this.alertType = true;
+          this.updateScoreBoard(currentIndex + 1, 'attempted', selectedOptionValue, this.currentScore);
+          this.correctFeedBackTimeOut();
+          if (!this.showFeedBack) {
+            this.nextSlide();
+          }
+          if (this.showFeedBack) {
+            this.updateScoreBoard(((currentIndex + 1)), 'correct', undefined, this.currentScore);
+          }
+        } else if (!Boolean(option.option.value.value == correctOptionValue)) {
+          this.showAlert = true;
+          this.alertType = false;
+          if (this.showFeedBack) {
+            this.updateScoreBoard((currentIndex + 1), 'wrong');
+          }
+          if (!this.showFeedBack) {
+            this.nextSlide();
+          }
         }
-        if (this.showFeedBack) {
-          this.updateScoreBoard(((currentIndex + 1)), 'correct' , undefined , this.currentScore);
-        }
-      } else if (!Boolean(option.option.value.value == correctOptionValue)) {
-        this.showAlert = true;
-        this.alertType = false;
-        if (this.showFeedBack) {
-          this.updateScoreBoard((currentIndex + 1), 'wrong');
-        }
-        if (!this.showFeedBack) {
-          this.nextSlide();
+      }
+      if (option.cardinality === 'multiple') {
+        let key:any  = await this.utilService.getKeyValue(Object.keys(this.questions[currentIndex].responseDeclaration));
```

Comment:
Here correct Response is not checked

Reply:
i did not understand , can you mention it more clearly